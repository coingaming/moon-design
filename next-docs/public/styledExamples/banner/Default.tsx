import { Banner } from '@heathmont/moon-core';

const Example = () => (
  <Banner title="Title here" description="Lorem ipsum dolor..." />
);

export default Example;
