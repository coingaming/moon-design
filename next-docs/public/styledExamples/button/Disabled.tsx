import { Button } from '@heathmont/moon-core';

const Example = () => <Button disabled>Disabled</Button>;

export default Example;
