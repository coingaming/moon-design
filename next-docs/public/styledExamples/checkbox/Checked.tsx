import { Checkbox } from '@heathmont/moon-core';
import { useState } from 'react';

const Example = () => {
  const [isChecked, setIsChecked] = useState(true);
  return (
    <Checkbox
      checked={isChecked}
      label="Checkbox with checked state by default"
      onClick={() => setIsChecked(!isChecked)}
    />
  );
};

export default Example;
