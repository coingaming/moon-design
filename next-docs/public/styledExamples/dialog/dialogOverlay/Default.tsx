import React from 'react';
import { DialogOverlay } from '@heathmont/moon-components';
import { Button } from '@heathmont/moon-core';

const Example = () => {
  const [showDialog, setShowDialog] = React.useState(false);

  return (
    <>
      <Button variant="tertiary" onClick={() => setShowDialog(true)}>
        Open Dialog
      </Button>
      <DialogOverlay
        isOpen={showDialog}
        onDismiss={() => setShowDialog(false)}
      />
    </>
  );
};

export default Example;
