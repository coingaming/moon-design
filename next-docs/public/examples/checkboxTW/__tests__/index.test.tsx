/**
 * @jest-environment jsdom
 */

import React from 'react';
import { moonDesignLight, ThemeProvider } from '@heathmont/moon-themes';
import renderer from 'react-test-renderer';
import Default from '../Default';
import WithLabel from '../WithLabel';
import Checked from '../Checked';
import NoLabel from '../NoLabel';
import Disabled from '../Disabled';
import ReadOnly from '../ReadOnly';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>{component}</ThemeProvider>
);

describe('Checkbox', () => {
  it('renders default', () => {
    const testRenderer = renderer.create(renderWithTheme(<Default />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with label', () => {
    const testRenderer = renderer.create(renderWithTheme(<WithLabel />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders checked', () => {
    const testRenderer = renderer.create(renderWithTheme(<Checked />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders without label', () => {
    const testRenderer = renderer.create(renderWithTheme(<NoLabel />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders disabled', () => {
    const testRenderer = renderer.create(renderWithTheme(<Disabled />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders read only', () => {
    const testRenderer = renderer.create(renderWithTheme(<ReadOnly />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
