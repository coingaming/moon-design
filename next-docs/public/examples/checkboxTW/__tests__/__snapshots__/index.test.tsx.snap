// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox in Dark Theme renders Checked 1`] = `
<div
  className="inline-block"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="flex items-center text-moon-14 text-bulma cursor-pointer "
    htmlFor="checked"
  >
    <span
      className="flex me-3 relative"
    >
      <span
        className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-piccolo/[.12]"
      />
      <input
        checked={true}
        className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer bg-piccolo shadow-none"
        id="checked"
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "backgroundImage": "url('data:image/svg+xml;utf8, %3Csvg%20width%3D%221em%22%20height%3D%221em%22%20viewBox%3D%220%200%2032%2032%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20style%3D%22color%3A%23ffffff%22%3E%3Cpath%20d%3D%22M7%2014.941L13.667%2022%2025%2010%22%20stroke%3D%22currentColor%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E')",
          }
        }
        type="checkbox"
      />
    </span>
    Checked
  </label>
</div>
`;

exports[`Checkbox in Dark Theme renders Default 1`] = `
<div
  className="inline-block"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="flex items-center text-moon-14 text-bulma cursor-pointer "
  >
    <span
      className="flex me-3 relative"
    >
      <span
        className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
      />
      <input
        checked={false}
        className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks"
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "backgroundImage": "",
          }
        }
        type="checkbox"
      />
    </span>
  </label>
</div>
`;

exports[`Checkbox in Dark Theme renders Disabled 1`] = `
Array [
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  opacity-30 cursor-not-allowed select-none"
      htmlFor="disabled1"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
        />
        <input
          checked={false}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks opacity-30 cursor-not-allowed select-none"
          disabled={true}
          id="disabled1"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "",
            }
          }
          type="checkbox"
        />
      </span>
      Disabled
    </label>
  </div>,
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  opacity-30 cursor-not-allowed select-none"
      htmlFor="disabled2"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-piccolo/[.12]"
        />
        <input
          checked={true}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer bg-piccolo shadow-none opacity-30 cursor-not-allowed select-none"
          disabled={true}
          id="disabled2"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url('data:image/svg+xml;utf8, %3Csvg%20width%3D%221em%22%20height%3D%221em%22%20viewBox%3D%220%200%2032%2032%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20style%3D%22color%3A%23ffffff%22%3E%3Cpath%20d%3D%22M7%2014.941L13.667%2022%2025%2010%22%20stroke%3D%22currentColor%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E')",
            }
          }
          type="checkbox"
        />
      </span>
      Disabled Checked
    </label>
  </div>,
]
`;

exports[`Checkbox in Dark Theme renders NoLabel 1`] = `
<div
  className="inline-block"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="flex items-center text-moon-14 text-bulma cursor-pointer "
    htmlFor="noLabel"
  >
    <span
      className="flex me-3 relative"
    >
      <span
        className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
      />
      <input
        checked={false}
        className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks"
        id="noLabel"
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "backgroundImage": "",
          }
        }
        type="checkbox"
      />
    </span>
  </label>
</div>
`;

exports[`Checkbox in Dark Theme renders ReadOnly 1`] = `
Array [
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  cursor-not-allowed select-none"
      htmlFor="readOnly1"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
        />
        <input
          checked={false}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks cursor-not-allowed select-none"
          id="readOnly1"
          onChange={[Function]}
          onClick={[Function]}
          readOnly={true}
          style={
            Object {
              "backgroundImage": "",
            }
          }
          type="checkbox"
        />
      </span>
      ReadOnly
    </label>
  </div>,
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  cursor-not-allowed select-none"
      htmlFor="readOnly2"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-piccolo/[.12]"
        />
        <input
          checked={true}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer bg-piccolo shadow-none cursor-not-allowed select-none"
          id="readOnly2"
          onChange={[Function]}
          onClick={[Function]}
          readOnly={true}
          style={
            Object {
              "backgroundImage": "url('data:image/svg+xml;utf8, %3Csvg%20width%3D%221em%22%20height%3D%221em%22%20viewBox%3D%220%200%2032%2032%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20style%3D%22color%3A%23ffffff%22%3E%3Cpath%20d%3D%22M7%2014.941L13.667%2022%2025%2010%22%20stroke%3D%22currentColor%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E')",
            }
          }
          type="checkbox"
        />
      </span>
      ReadOnly Checked
    </label>
  </div>,
]
`;

exports[`Checkbox in Dark Theme renders WithLabel 1`] = `
<div
  className="inline-block"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="flex items-center text-moon-14 text-bulma cursor-pointer "
    htmlFor="withLabel"
  >
    <span
      className="flex me-3 relative"
    >
      <span
        className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
      />
      <input
        checked={false}
        className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks"
        id="withLabel"
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "backgroundImage": "",
          }
        }
        type="checkbox"
      />
    </span>
    With label
  </label>
</div>
`;

exports[`Checkbox in Light Theme renders Checked 1`] = `
<div
  className="inline-block"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="flex items-center text-moon-14 text-bulma cursor-pointer "
    htmlFor="checked"
  >
    <span
      className="flex me-3 relative"
    >
      <span
        className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-piccolo/[.12]"
      />
      <input
        checked={true}
        className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer bg-piccolo shadow-none"
        id="checked"
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "backgroundImage": "url('data:image/svg+xml;utf8, %3Csvg%20width%3D%221em%22%20height%3D%221em%22%20viewBox%3D%220%200%2032%2032%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20style%3D%22color%3A%23ffffff%22%3E%3Cpath%20d%3D%22M7%2014.941L13.667%2022%2025%2010%22%20stroke%3D%22currentColor%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E')",
          }
        }
        type="checkbox"
      />
    </span>
    Checked
  </label>
</div>
`;

exports[`Checkbox in Light Theme renders Default 1`] = `
<div
  className="inline-block"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="flex items-center text-moon-14 text-bulma cursor-pointer "
  >
    <span
      className="flex me-3 relative"
    >
      <span
        className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
      />
      <input
        checked={false}
        className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks"
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "backgroundImage": "",
          }
        }
        type="checkbox"
      />
    </span>
  </label>
</div>
`;

exports[`Checkbox in Light Theme renders Disabled 1`] = `
Array [
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  opacity-30 cursor-not-allowed select-none"
      htmlFor="disabled1"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
        />
        <input
          checked={false}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks opacity-30 cursor-not-allowed select-none"
          disabled={true}
          id="disabled1"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "",
            }
          }
          type="checkbox"
        />
      </span>
      Disabled
    </label>
  </div>,
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  opacity-30 cursor-not-allowed select-none"
      htmlFor="disabled2"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-piccolo/[.12]"
        />
        <input
          checked={true}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer bg-piccolo shadow-none opacity-30 cursor-not-allowed select-none"
          disabled={true}
          id="disabled2"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url('data:image/svg+xml;utf8, %3Csvg%20width%3D%221em%22%20height%3D%221em%22%20viewBox%3D%220%200%2032%2032%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20style%3D%22color%3A%23ffffff%22%3E%3Cpath%20d%3D%22M7%2014.941L13.667%2022%2025%2010%22%20stroke%3D%22currentColor%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E')",
            }
          }
          type="checkbox"
        />
      </span>
      Disabled Checked
    </label>
  </div>,
]
`;

exports[`Checkbox in Light Theme renders NoLabel 1`] = `
<div
  className="inline-block"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="flex items-center text-moon-14 text-bulma cursor-pointer "
    htmlFor="noLabel"
  >
    <span
      className="flex me-3 relative"
    >
      <span
        className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
      />
      <input
        checked={false}
        className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks"
        id="noLabel"
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "backgroundImage": "",
          }
        }
        type="checkbox"
      />
    </span>
  </label>
</div>
`;

exports[`Checkbox in Light Theme renders ReadOnly 1`] = `
Array [
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  cursor-not-allowed select-none"
      htmlFor="readOnly1"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
        />
        <input
          checked={false}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks cursor-not-allowed select-none"
          id="readOnly1"
          onChange={[Function]}
          onClick={[Function]}
          readOnly={true}
          style={
            Object {
              "backgroundImage": "",
            }
          }
          type="checkbox"
        />
      </span>
      ReadOnly
    </label>
  </div>,
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  cursor-not-allowed select-none"
      htmlFor="readOnly2"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-piccolo/[.12]"
        />
        <input
          checked={true}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer bg-piccolo shadow-none cursor-not-allowed select-none"
          id="readOnly2"
          onChange={[Function]}
          onClick={[Function]}
          readOnly={true}
          style={
            Object {
              "backgroundImage": "url('data:image/svg+xml;utf8, %3Csvg%20width%3D%221em%22%20height%3D%221em%22%20viewBox%3D%220%200%2032%2032%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20style%3D%22color%3A%23ffffff%22%3E%3Cpath%20d%3D%22M7%2014.941L13.667%2022%2025%2010%22%20stroke%3D%22currentColor%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E')",
            }
          }
          type="checkbox"
        />
      </span>
      ReadOnly Checked
    </label>
  </div>,
]
`;

exports[`Checkbox in Light Theme renders WithLabel 1`] = `
<div
  className="inline-block"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="flex items-center text-moon-14 text-bulma cursor-pointer "
    htmlFor="withLabel"
  >
    <span
      className="flex me-3 relative"
    >
      <span
        className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
      />
      <input
        checked={false}
        className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks"
        id="withLabel"
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "backgroundImage": "",
          }
        }
        type="checkbox"
      />
    </span>
    With label
  </label>
</div>
`;

exports[`Checkbox in RTL renders Checked 1`] = `
<div
  dir="rtl"
>
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer "
      htmlFor="checked"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-piccolo/[.12]"
        />
        <input
          checked={true}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer bg-piccolo shadow-none"
          id="checked"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url('data:image/svg+xml;utf8, %3Csvg%20width%3D%221em%22%20height%3D%221em%22%20viewBox%3D%220%200%2032%2032%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20style%3D%22color%3A%23ffffff%22%3E%3Cpath%20d%3D%22M7%2014.941L13.667%2022%2025%2010%22%20stroke%3D%22currentColor%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E')",
            }
          }
          type="checkbox"
        />
      </span>
      Checked
    </label>
  </div>
</div>
`;

exports[`Checkbox in RTL renders Default 1`] = `
<div
  dir="rtl"
>
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer "
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
        />
        <input
          checked={false}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "",
            }
          }
          type="checkbox"
        />
      </span>
    </label>
  </div>
</div>
`;

exports[`Checkbox in RTL renders Disabled 1`] = `
<div
  dir="rtl"
>
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  opacity-30 cursor-not-allowed select-none"
      htmlFor="disabled1"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
        />
        <input
          checked={false}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks opacity-30 cursor-not-allowed select-none"
          disabled={true}
          id="disabled1"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "",
            }
          }
          type="checkbox"
        />
      </span>
      Disabled
    </label>
  </div>
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  opacity-30 cursor-not-allowed select-none"
      htmlFor="disabled2"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-piccolo/[.12]"
        />
        <input
          checked={true}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer bg-piccolo shadow-none opacity-30 cursor-not-allowed select-none"
          disabled={true}
          id="disabled2"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url('data:image/svg+xml;utf8, %3Csvg%20width%3D%221em%22%20height%3D%221em%22%20viewBox%3D%220%200%2032%2032%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20style%3D%22color%3A%23ffffff%22%3E%3Cpath%20d%3D%22M7%2014.941L13.667%2022%2025%2010%22%20stroke%3D%22currentColor%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E')",
            }
          }
          type="checkbox"
        />
      </span>
      Disabled Checked
    </label>
  </div>
</div>
`;

exports[`Checkbox in RTL renders NoLabel 1`] = `
<div
  dir="rtl"
>
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer "
      htmlFor="noLabel"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
        />
        <input
          checked={false}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks"
          id="noLabel"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "",
            }
          }
          type="checkbox"
        />
      </span>
    </label>
  </div>
</div>
`;

exports[`Checkbox in RTL renders ReadOnly 1`] = `
<div
  dir="rtl"
>
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  cursor-not-allowed select-none"
      htmlFor="readOnly1"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
        />
        <input
          checked={false}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks cursor-not-allowed select-none"
          id="readOnly1"
          onChange={[Function]}
          onClick={[Function]}
          readOnly={true}
          style={
            Object {
              "backgroundImage": "",
            }
          }
          type="checkbox"
        />
      </span>
      ReadOnly
    </label>
  </div>
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer  cursor-not-allowed select-none"
      htmlFor="readOnly2"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-piccolo/[.12]"
        />
        <input
          checked={true}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer bg-piccolo shadow-none cursor-not-allowed select-none"
          id="readOnly2"
          onChange={[Function]}
          onClick={[Function]}
          readOnly={true}
          style={
            Object {
              "backgroundImage": "url('data:image/svg+xml;utf8, %3Csvg%20width%3D%221em%22%20height%3D%221em%22%20viewBox%3D%220%200%2032%2032%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20style%3D%22color%3A%23ffffff%22%3E%3Cpath%20d%3D%22M7%2014.941L13.667%2022%2025%2010%22%20stroke%3D%22currentColor%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E')",
            }
          }
          type="checkbox"
        />
      </span>
      ReadOnly Checked
    </label>
  </div>
</div>
`;

exports[`Checkbox in RTL renders WithLabel 1`] = `
<div
  dir="rtl"
>
  <div
    className="inline-block"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <label
      className="flex items-center text-moon-14 text-bulma cursor-pointer "
      htmlFor="withLabel"
    >
      <span
        className="flex me-3 relative"
      >
        <span
          className="w-6 h-6 rounded-moon-i-xs duration-200 absolute top-1/2 left-1/2 translate-x-[-50%] translate-y-[-50%] transition-[background-color] hover:bg-trunks/[.12]"
        />
        <input
          checked={false}
          className="appearance-none h-4 w-4 rounded-moon-i-xs focus:outline-none transition duration-200 align-top bg-no-repeat bg-center bg-contain float-start cursor-pointer shadow-[0_0_0_1px_inset] shadow-trunks"
          id="withLabel"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "",
            }
          }
          type="checkbox"
        />
      </span>
      With label
    </label>
  </div>
</div>
`;
